cmd/nri-kubernetes/main.go

```diff
@@ -79,9 +79,19 @@ func main() {
 		defer ksmScraper.Close()
 	}
 
+	var controlplaneScraper *controlplane.Scraper
+	if c.ControlPlane.Enabled {
+		controlplaneScraper, err = setupControlPlane(c, clients)
+		if err != nil {
+			logger.Errorf("setting up control plane scraper: %w", err)
+			os.Exit(exitSetup)
+		}
+		defer controlplaneScraper.Close()
+	}
+
 	for {
 		// TODO think carefully to the signature of this function
-		err := runScrapers(c, ksmScraper, kubeletScraper, i, clients)
+		err := runScrapers(c, ksmScraper, kubeletScraper, controlplaneScraper, i, clients)
```

Comment:
could we already add an interface here and call the function against []scrapers?

Reply:
I missed your comment literally two lines below